spa/src/components/settings/Organization/Organization.tsx

```diff
@@ -69,11 +67,19 @@ const Organization = () => {
   const companyTaxStatus = watch('companyTaxStatus');
   const taxId = watch('taxId');
 
+  useEffect(() => {
+    // "showSuccess" should not be in the useEffect dependencies, otherwise it will always set to false
+    if (showSuccess) {
+      setShowSuccess(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [companyName, companyTaxStatus, taxId, setShowSuccess]);
+
```

Comment:
```suggestion
  useEffect(() => {
    setShowSuccess(false);
  }, [companyName, companyTaxStatus, taxId, setShowSuccess]);
```

I tested this locally and it seemed harmless to set showSuccess to false repeatedly. The other approach here that would mean you wouldn't need to have `showSuccess` as a dep is to use the functional setter, e.g. `setShowSuccess((old) => old ? false : old);` but I think it's effectively the same as just setting it to false in all cases.

Reply:
I wanted to avoid re-renders, but I just did a quick search and your suggestion doesn't re-render even if we call setstate repeated times 😄 
ref: https://dev.to/sunflower/reactjs-if-it-is-setting-a-state-with-the-same-value-will-the-component-be-re-rendered-5g24